crates/cdk/src/nuts/nut00.rs

```diff
@@ -123,11 +122,40 @@ pub struct BlindSignature {
     /// DLEQ Proof
     ///
     /// <https://github.com/cashubtc/nuts/blob/main/12.md>
-    #[serde(default)]
     #[serde(skip_serializing_if = "Option::is_none")]
     pub dleq: Option<BlindSignatureDleq>,
 }
 
+/// Witness
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Witness {
+    /// P2PK Witness
+    #[serde(serialize_with = "p2pk_witness_serialize")]
+    #[serde(deserialize_with = "p2pk_witness_deserialize")]
+    P2PKWitness(P2PKWitness),
+    /// HTLC Witness
+    #[serde(serialize_with = "htlc_witness_serialize")]
+    #[serde(deserialize_with = "htlc_witness_deserialize")]
```

Comment:
My personal preference is to have a module that has a `serialize` and `deserialize` function, which then can be used as `#[serde(with = “module_name”)]`. If you prefer this way, that's valid, but I just wanted to share if you also like that shorthand.